### PR TITLE
docs: update a reference link and doc workflow

### PR DESCRIPTION
**Pull request by rootsongjc**
- Update the `watching.md` link's title on page https://gateway.envoyproxy.io/design/system-design.html
- Lack of env settings before you make the doc

### CODE REVIEW

File: docs/dev/DOCS.md [hunk 1]
```diff
@@ -16,8 +16,16 @@ to be in `docs/index.rst`'s `toctree` though.
...
 ## Documentation Workflow
 
-To work with the docs, just edit reStructuredText or Markdown files in `docs`, 
-then run
+To work with the docs, just edit reStructuredText or Markdown files in `docs`.
+Before you compile the docs, you need to set the `ENVOY_VERSION` and 
+`GATEWAYAPI_VERSION` environment variables. For example:
+
+```bash
+export ENVOY_VERSION=1.23
```
Comment (Xunzhuo) [inline docs/dev/DOCS.md:24]: I do not think we need to manually export these env variables, these vars are computed (eg version -e) by code for now.

Comment (Xunzhuo) [reply docs/dev/DOCS.md:24]: Can you revert the changes in this file? Other changes LGTM. Thanks!

Comment (rootsongjc) [reply docs/dev/DOCS.md]: Done

**Review by Xunzhuo — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**